Doctrine/Orm/Filter/AbstractFilter.php

```diff
@@ -69,7 +72,111 @@ protected function getClassMetadata(ResourceInterface $resource)
      */
     protected function isPropertyEnabled($property)
     {
-        return null === $this->properties || array_key_exists($property, $this->properties);
+        $enabled = false;
+
+        if (null === $this->properties) {
+            // to ensure sanity, nested properties must still be explicitly enabled
+            $enabled = !$this->isPropertyNested($property);
+        } else {
+            $enabled = array_key_exists($property, $this->properties);
+        }
+
+        return $enabled;
+    }
+
+    /**
+     * Determines whether the given property is mapped.
+     *
+     * @param string            $property
+     * @param ResourceInterface $resource
+     * @param bool              $allowAssociation
+     *
+     * @return bool
+     */
+    protected function isPropertyMapped($property, ResourceInterface $resource, $allowAssociation = false)
+    {
+        $mapped = false;
+
+        if ($this->isPropertyNested($property)) {
+            $propertyParts = $this->splitPropertyParts($property);
+            $metadata = $this->getNestedMetadata($resource, $propertyParts['associations']);
+
+            $mapped = $metadata->hasField($propertyParts['field']);
+
+            if (!$mapped && $allowAssociation) {
+                $mapped = $metadata->hasAssociation($propertyParts['field']);
+            }
+        } else {
```

Comment:
IMO `if-else` is perfectly appropriate for branching (which is exactly what's happening here). I'd agree with you if it's checking of a precondition.